src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs

```diff
@@ -16,31 +16,29 @@ public void AddTokens<TController>(TestServerAction action, TestServerTokenColle
 
             for (var i = 0; i < parameters.Length; i++)
             {
-                if (!IgnoreHeader(parameters[i]))
+                if (parameters[i].ParameterType.IsPrimitiveType()
+                    && !IgnoreHeader(parameters[i]))
                 {
-                    if (IsPrimitiveType(parameters[i].ParameterType))
-                    {
-                        var tokenName = parameters[i].Name.ToLowerInvariant();
-                        var tokenValue = action.ArgumentValues[i].Instance;
+                    var tokenName = parameters[i].Name.ToLowerInvariant();
```

Comment:
Maybe you could extract this variable outside and share with all cases

Reply:
done